### PR TITLE
Add release notes for C/C++ caching

### PR DESCRIPTION
**Pull request by wolfs**

### CODE REVIEW

File: subprojects/docs/src/docs/release/notes.md [hunk 1]
```diff
@@ -62,6 +62,10 @@ This could avoid potential vulnerabilities.
 It is now possible to compile native applications with the Visual C++ toolchain packaged with all versions of Visual Studio 2017.
   Note that discovery of a Visual Studio 2017 installation requires the [vswhere utility](https://github.com/Microsoft/vswhere).  Visual Studio 2017 versions earlier than update 2 do not install `vswhere` automatically, and so to use one of these earlier versions of Visual Studio 2017 when `vswhere` is not installed, you'll need to set [the installation directory on the VisualCpp toolchain](userguide/native_software.html#sec:defining_tool_chains).   
 
+### C/C++ incremental build improvements
+
+C/C++ compilation now takes the compiler version and system headers into account, making it safer to use those tasks with incremental build and [experimental native caching](userguide/build_cache.html#sec:task_output_caching_native_tasks).
+
```
Comment (lptr) [inline subprojects/docs/src/docs/release/notes.md]: I think this is fine, but we could also give a bit of explanation on why it's important to track the compiler version.

File: subprojects/docs/src/docs/userguide/buildCache.adoc [hunk 2]
```diff
@@ -167,11 +167,7 @@ Caching is then enabled for api:org.gradle.language.c.tasks.CCompile[] and api:o
 
 [NOTE]
 ====
-Caching native tasks is experimental.
-Some inputs to the compilation tasks are not tracked yet, which can cause false cache hits:
-
-* The compiler type and version, thus object files produced by different compilers will be treated as identical.
-* System headers, thus object files produced with different versions of system libraries will be treated as identical.
+C/C++ compilers embed absolute paths to sources into object files, so you may not be able to debug object files which have been loaded from the cache.
```
Comment (lptr) [inline subprojects/docs/src/docs/userguide/buildCache.adoc]: Did you want to remove "Caching native tasks is experimental" as well?

Comment (wolfs) [reply subprojects/docs/src/docs/userguide/buildCache.adoc]: I thought about it. Given that we still have the system property that we need to activate it felt strange removing it. So I left it in. WDYT?

File: subprojects/docs/src/docs/release/notes.md (hunk 1 above)
Comment (wolfs) [reply subprojects/docs/src/docs/release/notes.md]: I added some explanations.

**Pull request merged**